docs/usage/string-pattern-matching.md

```diff
@@ -76,6 +108,9 @@ For example, the pattern `["/^abc/", "!/^abcd/", "!/abce/"]`:
 - matches `"abc"` and `"abcf"`
 - does _not_ match `"foo"`, `"abcd"`, `"abce"`, or `"abcdef"`
 
+If you find yourself in a situation where you need to positive-match a string which starts with `!`, then you need to do so using a regular expression pattern.
+For example, `["/^!abc/$"]` will positively match against the string `"!abc"`.
```

Comment:
```suggestion
For example, `["!/^abc$/"]` will positively match against the string `"!abc"`.
```
typo?

Reply:
Not a typo

Reply:
It's intended to show how to match a string which starts with ! so the pattern needs to be inside the / /

Reply:
but the dollar needs to be inside the regex